clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -2630,6 +2631,10 @@ static void CollectArgsForIntegratedAssembler(Compilation &C,
         CmdArgs.push_back("-massembler-no-warn");
       } else if (Value == "--noexecstack") {
         UseNoExecStack = true;
+      } else if (Value == "-mmsa") {
```

Comment:
This code block should be moved above to the `switch`. There is a `case llvm::Triple::mips:` for MIPS.

Reply:
OK, good idea, I applied it, Thanks!